toolkit/pallets/address-associations/src/lib.rs

```diff
@@ -0,0 +1,105 @@
+//! Pallet storing associations from main chain addresses to parter chain address.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+extern crate alloc;
+
+pub use pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use alloc::string::ToString;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::OriginFor;
+	use sidechain_domain::{
+		MainchainAddress, MainchainKeyHash, MainchainPublicKey, MainchainSignature, UtxoId,
+	};
+	// use frame_system::pallet_prelude::*;
+	// use scale_info::prelude::fmt::Debug;
+	use sp_address_associations::AddressAssociationSignedMessage;
+	// use sp_std::vec::Vec;
+
+	pub const PALLET_VERSION: u32 = 1;
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// Partner chain address type
+		type PartnerChainAddress: Member + Parameter + MaxEncodedLen;
+
+		fn genesis_utxo() -> UtxoId;
+	}
+
+	#[pallet::storage]
+	pub type AddressAssociations<T: Config> = StorageMap<
+		Hasher = Blake2_128Concat,
+		Key = MainchainAddress,
+		Value = T::PartnerChainAddress,
+		QueryKind = OptionQuery,
+	>;
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::call_index(0)]
+		#[pallet::weight((2, DispatchClass::Normal))]
```

Comment:
it is not necessary to provide DispatchClass if it is normal. Also please do not forget to benchmark this